Assets/Scripts/GenerateMap.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+public class GenerateMap : MonoBehaviour
+{
+    [SerializeField] private Camera mainCamera;
+    [SerializeField] private GameObject NodePrefab;
+    [SerializeField] private GameObject RootNode;
+    [SerializeField] private GameObject EndNode;
+    [SerializeField] private int rows;
+    [SerializeField] private float nodeSpacing = 1f;
+    [SerializeField] private int ymin;
```

Comment:
nitpick, probably could be like that for consistency

```suggestion
    [SerializeField] private float ymin;
```